marimo/_output/formatters/npndarray_formatters.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+from marimo._messaging.mimetypes import KnownMimeType
+from marimo._output.formatters.formatter_factory import FormatterFactory
+
+
+class NpndarrayFormatter(FormatterFactory):
+    @staticmethod
+    def package_name() -> str:
+        return "np.ndarray class type"
+
+    def register(self) -> None:
+        from marimo._output import formatting
+        import numpy as np
+
+        @formatting.formatter(np.ndarray)
+        def _show_npndarray_plot(plot: np.ndarray) -> tuple[KnownMimeType, str]:
+            import matplotlib.pyplot as plt
+
+            # Ensure that the plot is shown within the notebook
+            plt.show()
```

Comment:
I tried out a very small example - it does return a numpy array. but we don't know whats inside of it. (it could be numbers, or matplotlib in this case). But we can't always do `plt.show()`

![image](https://github.com/marimo-team/marimo/assets/2753772/bd371779-d062-4ddb-be15-0afb7b90146a)

something we could explore:

1. unpack nparray's like we do with the list and dicts formatter. 
	- This could be expensive for large nparrays. So maybe we can put a limit (e.g. 10)
	- This could also maybe be recursive, so we would need to avoid an infinite loop
2. We could add our own plotting backend https://python.arviz.org/en/stable/contributing/plotting_backends.html#. We do this with matplotlib

Reply:
Will look into this. I'm supposing that the current "fix" isn't ideal and could be transformed into something related to how matplotlib plots are handled overall in marimo.

Saw the `matplotlib_formatters` in marimo\_output\formatters and the `mpl` file as well. Should I try and approach similar to that?

Reply:
@Haleshot -- I saw your discussion in the issue. Assuming it works, I think it would be simpler to ask the user to call plt.show() or plt.gca() rather than introspect numpy arrays. Or maybe there's a way to do it with a custom backend as Myles mentioned.

Have you checked to see how Arviz is rendered in IPython notebooks? Perhaps that could serve as inspiration

Reply:
> @Haleshot -- I saw your discussion in the issue. ~Assuming it works, I think it would be simpler to ask the user to call plt.show() or plt.gca() rather than introspect numpy arrays.~ Maybe there's a way to do it with a custom backend as Myles mentioned.
> 
> Have you checked to see how Arviz is rendered in IPython notebooks? Perhaps that could serve as inspiration

I wanted to ask if there's any particular source for IPython thaat you refer to (found the official GitHub [here](https://jupyter-notebook.readthedocs.io/en/stable/index.html) and for jupyter [here])(https://jupyter-notebook.readthedocs.io/en/stable/index.html). (Also tried asking in the discord thread).

Also apologize if I'm prolonging an easy fix, making it hard for you both to review and come back to every time.